### ISSUE TITLE
Add "Help" icon to modal design guidelines

### ISSUE THREAD

**Issue by mmenestr** [patternfly/patternfly-org#2194]
Can now add a help icon at the top right of a modal

**Comment by doruskova** [comment]
Document for a review [here](https://docs.google.com/document/d/13RWveUTqmv7jsowYuSacyyBHvBNvm3PY2vLXnElBvU4/edit?usp=sharing)

Don't know if it's enough to explain it in 3 sentences.

**Comment by mmenestr** [comment]
@doruskova I think that looks good! The only thing i'd maybe add is some example use cases of when you'd want to do use one general popover in the modal vs just using it in a specific form field for example. Not sure what drove the initial ask behind this design addition but that might help! @mcarrano what do you think?

**Comment by mcarrano** [comment]
I don't feel strongly either way about this. The ask came from the Cockpit team. @garrett might be able to shed some light on why they are using help at the modal level.

**Comment by garrett** [comment]
Thanks for adding the help icon on the dialog!

We did have some UX studies where people were a bit confused about general things, such as network bonds. We found that they needed further explanation at the level of the dialog.

Here's a screenshot crop showing the implementation in Cockpit:
![Screenshot_2020-11-16 Networking - garrett Rain](https://user-images.githubusercontent.com/10246/99284776-fba26100-2836-11eb-9d49-7c8a1a93e681.png)

I think @andreasn might remember specifics better than I do.

**Comment by andreasn** [comment]
> The only thing i'd maybe add is some example use cases of when you'd want to do use one general popover in the modal vs just using it in a specific form field for example. Not sure what drove the initial ask behind this design addition but that might help!

This should be used when a person using a product needs a short explanation of a new concept and, if needed, have easy access to more documentation.

As @garrett said, this came out of a [usability test we did in 2018](https://github.com/cockpit-project/cockpit/wiki/Usability-testing-of-various-functions) with 21 participants, half of them Linux admins and half of them Windows admins.
The specific issue with network bonding that the help popup tries to address was filed as [Cockpit issue 842](https://github.com/cockpit-project/cockpit/issues/842)

What we had to solve at the time was twofold. One part was that network bonding is a term that was familiar to the Linux admins, but not at all to the Windows admins (the closest in Windows Server is [Network Teaming](https://docs.microsoft.com/en-us/windows-server/networking/technologies/nic-teaming/nic-teaming)), hence we needed a short explanation of the concept. The second part is that Linux network bonding has very specific terminology for the different modes (balance-rr, balance-xor, 802.3ad etc.). We made a couple of mockups where we experimented with embedding the document directly, but it ended up being big blocks of text, due to fairly complex concepts, that duplicated what we had in the [product documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_systems_using_the_rhel_8_web_console/configuring-network-bonds-using-the-web-console_system-management-using-the-rhel-8-web-console) already, so we thought it made more sense to point them to the documentation directly.

We found in follow-up studies that having easy access to documentation helped a lot when trying to resolve hard problems like these.
